app/src/main/java/ingsw/group1/findmyphone/contacts/GenericContact.java

```diff
@@ -0,0 +1,99 @@
+package ingsw.group1.findmyphone.contacts;
+
+import androidx.annotation.NonNull;
+import androidx.room.ColumnInfo;
+import androidx.room.Ignore;
+import androidx.room.PrimaryKey;
+
+import ingsw.group1.msglibrary.Peer;
+
+/**
+ * Abstract class represents a generic contact
+ * A contact is identify by an address phone
+ * and it can have a name.
+ *
+ * @param <P> The type of peer that contains a valid phone address.
+ *
+ * @author Giorgia Bortoletti
+ */
+public abstract class GenericContact<P extends Peer<String>> {
```

Comment:
I'd use an Interface personally, multiple interfaces can be implemented, but only one class can be extended, so, if it's not strictly necessary the logic should all be inside the implementing class.
If you agree the interface can become `GenericContact<A, P extends Peer<A>>`

Reply:
We tried to use abstract classes for kademlia's nodes and we ended up switching to interfaces, this is similar